### ISSUE TITLE
Module does not support well-behaved reverse lookup under systemd-resolved

### ISSUE THREAD

**Issue by tpdownes** [hashicorp/terraform-aws-consul#155]
The current state of consul support for `systemd-resolved` is a bit funky in that forward resolution works well and simply, while reverse lookup of IPs will, with low percentage, return `.consul` domains or whatever the other DNS resolvers say. e.g. https://github.com/hashicorp/consul/issues/6462.

https://github.com/hashicorp/consul/pull/6731 provides a solution for ensuring that all reverse lookups of an IP address known to consul results in a `.consul` domain.

It has the added behavior (perhaps undesirable) of reverse lookup failing on IP addresses not known to consul unless one also configures the `recursors` option. e.g.
```
{
...
    "recursors": ["10.0.0.2"],
...
}
```
I think it would be reasonable to consider adopting this configuration in the consul-cluster module and intend this issue to be a starting point for the conversation.

Thanks for the module as it has facilitated very rapid progress on my side!

**Comment by brikis98** [comment]
Thanks for reporting!

> It has the added behavior (perhaps undesirable) of reverse lookup failing on IP addresses not known to consul 

What's the default behavior?

> unless one also configures the recursors option. e.g.

What value would you plug into `recursors`?

**Comment by tpdownes** [comment]
The behavior of the documented solution is described in https://github.com/hashicorp/consul/issues/6462.

TL;DR: most of the time reverse resolution goes through whatever the system has been configured to use. A small percentage of the time reverse resolution goes through consul. That is to say, reverse lookup is not predictable in the documented `systemd-resolved` solution.

The PR is predictable in that 100% of the time `systemd-resolved` will reverse lookup via the consul agent. The cost is that the consul agent will fail to reverse lookup for IPs not within the `.consul` domain.

You can solve this by placing a DNS server with appropriate reverse lookup capability into `recursors`. In most situations, this probably means whatever DNS server you use for forward lookups outside of the `.consul` domain. i.e., whatever DNS servers you got from DHCP / static config.

Make sense?

**Comment by brikis98** [comment]
Got it, thanks for the context. So, to summarize, you are proposing that for systems using `systemd-resolved` we update the `run-consul` script to be able to add the following to `resolved.conf`:

```bash
DNS=127.0.0.1
Domains=~consul ~<CIDR>.in-addr.arpa
```

Where `<CIDR>` is passed in via a new param to the script... As well as add the following to the consul config:

```hcl
    "recursors": ["<DNS_SERVER>"],
```

Where `<DNS_SERVER>` is also passed in via a new param to the script.

Is that right?

**Comment by tpdownes** [comment]
I'm definitely proposing the first thing, with caveat that it's actually the "backwards truncated CIDR" and it should be opt-in.

The second thing, I'm outlining the pros/cons. You could imagine mimicking the `dnsmasq` behavior of using servers from `/etc/resolve.conf` by having `run-consul` automatically set recursors  by parsing the output of `systemd-resolve` (or `resolvectl` on ultra-contemporary systems) or `networkctl`. I'm not really what is right these days.

**Comment by brikis98** [comment]
(sorry for delay, we were all away for a company offsite)

Roger. I think if both options are opt-in based on passed-in params, this makes sense to add. A PR is very welcome!
